.changeset/tame-eggs-notice.md

```diff
@@ -0,0 +1,56 @@
+---
+'@openfn/language-kobotoolbox': major
+---
+
+Improved pagination support and restricted HTTP methods to KoboToolbox API
+
+### Breaking Changes
+
+- Updated `getSubmission` pagination behavior:
+
+  - Removed `paginate` option
+  - Added new pagination options:
+    - `limit`: Controls number of results returned
+    - `pageSize`: Number of items per page
+
+- Remove export of `http` methods from `@openfn/language-common`
+- Add support for `maxRedirections` option in `HTTPRequestOptions`
+
+### Migration Guide
+
+1. Pagination Changes:
+
+   - Replace `paginate: true/false` with `limit` and `pageSize` options
+   - Default limit is `10000` if not specified
+
+   ```js
+   // Fetch submissions with default limit (10000)
+   getSubmission('form-id');
+
+   // Fetch all submissions (no limit)
+   getSubmission('form-id', { limit: Infinity });
+
+   // Fetch submissions with max results
+   getSubmission('form-id', { limit: 500 }); // Returns up to 500 submissions
+   ```
+
+2. HTTP Changes:
+
+- For KoboToolbox API requests: Continue using the built-in methods
```

Comment:
Was anyone using general purpose requests? I'm a bit suspicious of this.

I consider `http.get('https://any-api.com/endpoint');` to be invalid, undocumented behaviour. We don't tell anyone to do this (right??), and don't document it. So we can just remove this behaviour without making a big deal about it. The http namespace is also very new so not many users should be affected.

By documenting the breaking change like this, we sort of imply that the old behaviour was supported.

I wouldn't even report it as a bug fix or feature change or anything. I'd just say (when we're ready) `http methods will now throw an error when passing absolute URLs`. No other code examples, workarounds or docs are needed.

Reply:
In v2 we don't have any job that use `http.get()` to access a different rest api. 
Regarding removing `http` export from common from v1.0.4 to v2.4.3 `http` method was exported from common.
So a request like this would work 
```js
http.get({ url: 'https://jsonplaceholder.typicode.com/todos' })
```

In v3.0.0 we introduces kobotoolbox `http` methods. So this means we can't make request outside kobotoolbox
```js
http.get('https://any-api.com/endpoint');
```
But the `http` common export was not removed.

In this PR, i have removed the common export of http from common

I have removed the example in changeset and left the one line description for http changes